force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -102,16 +113,18 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer countVowels(String str) {
         // Initialize the result Integer
-
-        /*
-        for () {
-            // get individual characters from the string - hint use substring or split
-            // check if the character is a vowel
-            // if it is a vowel, increment the result Integer
+        Integer result = 0;
+        Set<String> vowels = new Set<String>{'a', 'e', 'i', 'o', 'u'};
+        str = str.toLowerCase();
+        for(Integer i = 0; i < str.length(); i++ ){
+            String c = str.substring(i, i+1);
+            if(vowels.contains(c)){
+                result++;
+                
```

Comment:
Good job. There are a few ways to solve this.  Again you don't really need to create the variable in memory on each iteration. My solution is below

```
  public static Integer countVowels(String str) {
        Integer vowelCount = 0;
        Set<String> vowels = new Set<String>{
                'a', 'e', 'i', 'o', 'u'
        };
        List<String> characters = str.split('');
        for (String character : characters) {
            if (vowels.contains(character.toLowerCase())) {
                vowelCount++;
            }
        }
        return vowelCount;
    }
    ```